src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,203 @@
+        if (!fs.existsSync(reviewPath)) {
+            fs.mkdir(reviewPath, () =>{
+                this.mapVideo(saveFrame, "last").then( () => {
+                    reviewModel();
+                });
+            });
+        } else {
+            reviewModel();
+        }
+
+        function reviewModel() {
+            //run the model on the reviewPath directory
+            self.detectionAlgorithmManager.setReviewer(method, modelPath);
+            self.detectionAlgorithmManager.reviewer.reviewImagesFolder(reviewPath).then( (modelTags) => {
+                self.videotagging.frames = [];
+                self.videotagging.optionalTags.createTagControls(Object.keys(modelTags.classes));
+
+                //Create regions based on the provided modelTags
+                Object.keys(modelTags.frames).map( (pathId) => {
+                    var frameImage = new Image();
+                    frameImage.src = path.join(reviewPath, pathId);
+                    frameImage.onload = loadFrameRegions; 
+
+                    function loadFrameRegions() {
+                        var imageWidth = this.width;
+                        var imageHeight = this.height;
+                        frameId = pathId.replace(".jpg", "");//remove.jpg
+                        self.videotagging.frames[frameId] = [];
+                        modelTags.frames[pathId].regions.forEach( (region) => {
+                            self.videotagging.frames[frameId].push({
+                            x1:region.x1,
+                            y1:region.y1,
+                            x2:region.x2,
+                            y2:region.y2,                          
+                            id:self.videotagging.uniqueTagId++,
+                            width:imageWidth,
+                            height:imageHeight,
+                            type:self.videotagging.regiontype,
+                            tags:Object.keys(modelTags.classes).filter( (key) => {return modelTags.classes[key] === region.class }),
+                            name:(self.videotagging.frames[frameId].length + 1),
+                            blockSuggest: true
+                            }); 
+                        });
+                    }
+                });
+                self.videotagging.showAllRegions();
+                //cleanup and notify
+                self.videotagging.video.currentTime = 0;
+                self.videotagging.playingCallback();
+                let notification = new Notification('Offline Video Tagger', { body: 'Model Ready For Review.' });
+                cb();
+            });
+        }
+
+        function saveFrame(frameId, fCanvas, canvasContext, saveCb){
+            var writePath =  path.join(reviewPath, `{frameId}.jpg`);
+            //write canvas to file and change frame
+            console.log('saving file', writePath);
+            if (!fs.exists(writePath,saveCb)) {
```

Comment:
This is an issue - you call fs.exists which is async so the return value will always evaluate to false.
The fix is to have fs.writeFile in the callback of fs.exists